### PR TITLE
Roles for devops tools

### PR DESCRIPTION
**Pull request by tself**
- Updated pip role to properly set umask so it doesn't require root to run everything
- Added ansible role
- Added azure-cli role
- Added Google Cloud SDK role
- Added terraform role
- Added vim role

### CODE REVIEW

File: ansible/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+ansible
+===
+
+An Ansible role to install Ansible.
```
Comment (tequalsme) [inline ansible/README.md]: state that Ansible will be installed via pip

File: ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Requirements
+------------
+
+Assumes RedHat OS family and access to the Internet (or access to local Pypi servers)
+
+Role Variables
+--------------
+
+    version: 2.6.5
```
Comment (tequalsme) [inline ansible/README.md]: I'd prefix this to `ansible_version`, variables can be set in a group_vars file if they are prefixed to the role. If they are named generically they must be passed directly on the role.

File: ansible/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+# tasks file for ansible
+
+- name: Install Ansible
+  pip:
+    name: "ansible=={{ version }}"
+    umask: "0022"
```
Comment (tequalsme) [inline ansible/tasks/main.yml]: I had no idea this option existed - beautiful!

File: aws_cli/tasks/main.yml [hunk 4]
```diff
@@ -2,4 +2,7 @@
 # tasks file for aws_cli
 
 - name: Install awscli
-  pip: name=awscli
+  pip:
```
Comment (tequalsme) [inline aws_cli/tasks/main.yml]: Ditto my comment for Ansible - state in the README that it will be installed via pip

File: pip/README.md [hunk 5]
```diff
@@ -11,6 +11,7 @@ Assumes RedHat OS family, and internet connectivity
 Role Variables
 --------------
 
+    install_with_script: no
```
Comment (tequalsme) [inline pip/README.md]: Describe the different behavior when `install_with_script` is enabled/disabled

**Comment by tequalsme** [comment]
Overall looks good, most of my comments are README related

File: packer/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+# defaults file for packer
+
+packer_zip_url: https://releases.hashicorp.com/packer/1.4.2/packer_1.4.2_linux_amd64.zip
```
Comment (tself) [inline packer/defaults/main.yml]: Matt suggested changing this to just packer_version and deriving the URL

File: terraform/defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+# defaults file for terraform
+
+tf_zip_url: https://releases.hashicorp.com/terraform/0.11.14/terraform_0.11.14_linux_amd64.zip
```
Comment (tself) [inline terraform/defaults/main.yml]: Matt suggested changing this to tf_version and deriving URL

**Pull request merged**